packages/devtools_app/lib/src/memory/flutter/memory_chart.dart

```diff
@@ -369,17 +790,52 @@ class SelectedDataPoint extends LineChartMarker {
     final num rss = values.rss.toDouble();
     final bool isGced = values.isGC;
 
-    final TextPainter painter = PainterUtils.create(
-      null,
-      'Time       ${_timestampFormatter.getFormattedValue1(timestampAsInt.toDouble())}\n'
-      'Capacity ${_formatter.getFormattedValue1(heapCapacity)}\n'
-      'Used       ${_formatter.getFormattedValue1(heapUsed)}\n'
-      'External  ${_formatter.getFormattedValue1(external)}\n'
-      'RSS        ${_formatter.getFormattedValue1(rss)}\n'
-      'GC          $isGced',
-      textColor,
-      fontSize,
-    )..textAlign = TextAlign.left;
+    // Alpha filled stacked:
+    final num memorySystemAndOther = values.memoryInfo.system.toDouble() +
+        values.memoryInfo.other.toDouble(); // Purple-ish
+    final num memoryCode = values.memoryInfo.code.toDouble(); // Gray Purple
+    final num memoryNativeHeap =
+        values.memoryInfo.nativeHeap.toDouble(); // Blue-ish
+    final num memoryJavaHeap =
+        values.memoryInfo.javaHeap.toDouble(); // Green-ish
+    final num memoryStack = values.memoryInfo.stack.toDouble(); // White-ish
+    final num memoryGraphics = values.memoryInfo.graphics.toDouble(); // Orangy
+
+    final num memoryTotal = values.memoryInfo.total.toDouble(); // dashed line
+
+    final TextPainter painter = type == ChartType.DartHeaps
+        ? PainterUtils.create(null, _titlesDartVm, textColor, fontSize)
+        : PainterUtils.create(null, _titlesAndroid, textColor, fontSize);
+
+    painter.textAlign = TextAlign.left;
+
+    // Compute the values of each point for a particular x position (timestamp).
+    final TextPainter painterValues = type == ChartType.DartHeaps
+        ? PainterUtils.create(
+            null,
+            '${_timestampFormatter.getFormattedValue1(timestampAsInt.toDouble())}\n'
+            '${_formatter.getFormattedValue1(heapCapacity)}\n'
+            '${_formatter.getFormattedValue1(heapUsed)}\n'
+            '${_formatter.getFormattedValue1(external)}\n'
+            '${_formatter.getFormattedValue1(rss)}\n'
+            '$isGced',
+            textColor,
+            fontSize,
+          )
+        : PainterUtils.create(
+            null,
+            '${_timestampFormatter.getFormattedValue1(timestampAsInt.toDouble())}\n'
```

Comment:
this line is duplicated. Pull into local var

Reply:
Done.